File: lib/tailwind/create_tw_spacing.dart
import 'dart:io' show File, Platform;
import 'package:path/path.dart' show dirname, join;
import 'dart:convert' show json;

void main() {
  String dir = join(dirname(Platform.script.path), 'spacing.json');
  String saveDir =
      join(dirname(Platform.script.path), '../core/tw_spacing.dart');
  Map data = json.decode(File(dir).readAsStringSync());

  String core = '';
  data.forEach((key, value) {
    String str =
        '\n/// ```\n/// left: $value;\n/// right: $value;\n/// ```\nstatic EdgeInsets x$key = const EdgeInsets.symmetric(horizontal: $value);\n';
    str +=
        '\n/// ```\n/// top: $value;\n/// bottom: $value;\n/// ```\nstatic EdgeInsets y$key = const EdgeInsets.symmetric(vertical: $value);\n';

    str +=
        '\n/// ```\n/// right: $value;\n/// ```\nstatic EdgeInsets r$key = const EdgeInsets.only(right: $value);\n';
    str +=
        '\n/// ```\n/// left: $value;\n/// ```\nstatic EdgeInsets l$key = const EdgeInsets.only(left: $value);\n';
    str +=
        '\n/// ```\n/// top: $value;\n/// ```\nstatic EdgeInsets t$key = const EdgeInsets.only(top: $value);\n';
    str +=
        '\n/// ```\n/// bottom: $value;\n/// ```\nstatic EdgeInsets b$key = const EdgeInsets.only(bottom: $value);\n';
    str +=
        '\n/// ```\n/// all: $value;\n/// ```\nstatic EdgeInsets a$key = const EdgeInsets.all($value);\n';

    core += str;
  });

  String result =
      'import "package:flutter/material.dart";\n\nclass TwSpacing {\n$core\n}';

  File(saveDir).writeAsStringSync(result);
}

File: lib/tailwind/create_tw_spacing.js
const data = require("./spacing.json");

let core = "";

for (let key in data) {
  let str = `/// \`\`\`
  /// left: ${data[key]};
  /// right: ${data[key]};
  /// \`\`\`
  static EdgeInsets x${key} = const EdgeInsets.symmetric(horizontal: ${data[key]});

  /// \`\`\`
  /// top: ${data[key]};
  /// bottom: ${data[key]};
  /// \`\`\`
  static EdgeInsets y${key} = const EdgeInsets.symmetric(vertical: ${data[key]});

  /// \`\`\`
  /// left: ${data[key]};
  /// \`\`\`
  static EdgeInsets l${key} = const EdgeInsets.only(left: ${data[key]});

  /// \`\`\`
  /// right: ${data[key]};
  /// \`\`\`
  static EdgeInsets r${key} = const EdgeInsets.only(right: ${data[key]});

  /// \`\`\`
  /// top: ${data[key]};
  /// \`\`\`
  static EdgeInsets t${key} = const EdgeInsets.only(top: ${data[key]});

  /// \`\`\`
  /// bottom: ${data[key]};
  /// \`\`\`
  static EdgeInsets b${key} = const EdgeInsets.only(bottom: ${data[key]});

  /// \`\`\`
  /// all: ${data[key]}
  /// \`\`\`
  static EdgeInsets a${key} = const EdgeInsets.all(${data[key]});\n
  `;

  core += str;
}

const str = `import 'package:flutter/material.dart';\n\nclass TwSpacing {
  ${core}
}`;

require("fs").writeFileSync("../core/tw_spacing.dart", str);
